hooks.go

```diff
@@ -136,18 +136,18 @@ func (h *Hooks) Provides(b ...byte) bool {
 
 // Add adds and initializes a new hook.
 func (h *Hooks) Add(hook Hook, config any) error {
-	h.Lock()
```

Comment:
Still needs a lock here for Add to prevent concurrent Add. Load/Store is atomic, but operations between them are not.

Reply:
Yes, I wondered about this as I was pushing it. This makes a lot more sense. Still, a little write lock is much better than RW locking.